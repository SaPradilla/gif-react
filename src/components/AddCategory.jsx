import { useState } from "react"
import PropTypes from "prop-types";


export const AddCategory = ({onNewValue})=>{

    const [inputValue,setInputValue] = useState()
    
    const onInputChange = (e)=>{
        setInputValue( e.target.value )    
    }

    const onSubmit = (e)=>{
        e.preventDefault();

        if ( inputValue.trim().length <= 1) return;
        onNewValue( inputValue.trim() );
        setInputValue('');

    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">

            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
            <button type="submit">Agregar</button>


        </form>        
    )

}

AddCategory.propTypes = {

    onNewValue : PropTypes.func.isRequired,

}