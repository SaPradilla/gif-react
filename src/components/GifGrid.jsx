import { useState,useEffect } from "react";
import GifItem from "./GifItem";

import { getGifs } from "../helpers/getGifs"
import useFetchGifs from "../hooks/useFetchGifs";

export const GifGrid = ({category})=>{

    const { images,isLoading } = useFetchGifs( category );

    return(
        <>
            <h3> { category }</h3>
            {
                isLoading && <h2> Cargando... </h2>
            }
            <div className="card-grid">
                {images.map( ( image) => (

                    <GifItem 
                        key={image.id}
                        title={ image.title}
                        url={image.url}
                    />

                ))}

            </div>
        </>
    )
}

// 3177403682