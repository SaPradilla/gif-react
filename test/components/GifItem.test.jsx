import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas de GitItems', () => {
  
  const title = 'Saitama';
  const url = 'http://imagen.com/image.jpg'

  test('debe de hacer match con el snapshot', () => {
    
    const { container } = render( <GifItem title={title} url={url} />)    
    
    expect( container ).toMatchSnapshot();


  })
  
  test('debe de mostrar la imagen con el url y el alt indicado ', () => {
    
    render( <GifItem title={title} url={url} /> );
    // imprimir el componente
    // screen.debug();
    const {src,alt} = screen.getByRole('img')
    
    expect ( src ).toBe( url );
    expect ( alt ).toBe( title );

  })
  
  test('debe de mostrar el titulo en el componente', () => {
    
    render( <GifItem title={title} url={url}/> );

    expect( screen.getByText( title ) ).toBeTruthy();

  })
  


})
