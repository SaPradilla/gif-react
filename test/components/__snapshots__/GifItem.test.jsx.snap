// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GitItems debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://imagen.com/image.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
